query-connector/src/app/queryBuilding/buildFromTemplates/ConditionColumnDisplay.tsx

```diff
@@ -56,9 +66,36 @@ export const ConditionColumnDisplay: React.FC<ConditionColumnDisplayProps> = ({
       name: prevValues.name,
       include: !prevValues.include,
     };
+    const shouldRemove = prevFetch[category][conditionId].include == false;
+    updateSelectedConditions(shouldRemove, category, conditionId, prevFetch);
     setFetchedConditions(prevFetch);
   }
 
+  const updateSelectedConditions = (
+    shouldRemove: boolean,
+    category: string,
```

Comment:
do we want to preserve at the category level? or even have a JSON versus an array?

I think we would rather preserve just at the `conditionId` level as the keys since we don't actually need categories for the next page or the name of the condition, but just the `conditionId` to do a look up to display. We could plausibly do that with just an array and pop/append as needed.

I don't think there's any harm in how it is now though; I think it could even be helpful if we want to ensure on the next page the conditions are "grouped" by category, even if it's not displayed.

Reply:
yeah, part of my thinking here was knowing we are tweaking how all the query data gets stored in the db - so wanting to preserve all the info we have now and we can always ease up if/when we decide we don't need it to be either stored in the db or shown in the UI. happy to rework later on if we decide it's too complicated/unnecessary here!